is_diag.hhs

```diff
@@ -0,0 +1,36 @@
+/**
+ * @author Jing Stone
+ * @param input - 1d list or 2d matrix
+ * @returns - 
+ *        Determine if matrix is diagonal, matrix could be rectangular.
+ */
+
+function is_diag(input) {
+   *import math: ndim
+
+  // true for []
+  if (input.length === 0) return true;
+  let dims = ndim(input)
+
+  // for 1d list, checkt if the element is 0 exclude input[0]
+  if (dims === 1) {
+    for (let i = 1; i < input.length; i++) {
+      if (input[i] !== 0) {
+        return false;
+      }
+    }
+    return true;
+  }
+  // check other elements except main diagonal are zero or not.
+  else {
```

Comment:
Hey so we still are supporting Mat objects, just at the 2d level. That mightve been mixed up in discord, sorry about that. So we still do the typical `in_type = (input instanceof Mat); raw_in = (in_type) ? input.clone() : deep_copy(input);` at the top with a `*import math: deep_copy` above it in the function. and so on, just like the other functions that support Mats.

I apologize for the confusion, I'll make it clear in the discord. Other than that it looks great.